CMake could not find OpenSSL on OS X 10.11

I think this is because OS X is moving away from OpenSSL... not sure though.

```sh
λ cargo new libssh2-sys-test
λ cd libssh2-sys-test
λ echo '\n[dependencies]\nlibssh2-sys = "0.1.34"\n' >> Cargo.toml
λ cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling gcc v0.3.20
   Compiling pkg-config v0.3.6
   Compiling winapi v0.2.5
   Compiling winapi-build v0.1.1
   Compiling libc v0.2.2
   Compiling ws2_32-sys v0.2.1
   Compiling openssl-sys v0.7.1
   Compiling libz-sys v1.0.0
   Compiling cmake v0.1.11
   Compiling libssh2-sys v0.1.34
failed to run custom build command for `libssh2-sys v0.1.34`
Process didn't exit successfully: `/Users/brendan/Desktop/libssh2-sys-test/target/debug/build/libssh2-sys-d5401c54977098a6/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/Users/brendan/.multirust/toolchains/stable/cargo/registry/src/github.com-0a35038f75765ae4/libssh2-sys-0.1.34/libssh2" "-DCRYPTO_BACKEND=OpenSSL" "-DBUILD_SHARED_LIBS=OFF" "-DENABLE_ZLIB_COMPRESSION=ON" "-DCMAKE_INSTALL_LIBDIR=lib" "-DBUILD_EXAMPLES=OFF" "-DBUILD_TESTING=OFF" "-DCMAKE_INSTALL_PREFIX=/Users/brendan/Desktop/libssh2-sys-test/target/debug/build/libssh2-sys-d5401c54977098a6/out" "-DCMAKE_C_FLAGS= -O0 -ffunction-sections -fdata-sections -m64 -fPIC" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Debug"
-- The C compiler identification is AppleClang 7.0.2.7000181
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring incomplete, errors occurred!
See also "/Users/brendan/Desktop/libssh2-sys-test/target/debug/build/libssh2-sys-d5401c54977098a6/out/build/CMakeFiles/CMakeOutput.log".

--- stderr
CMake Error at /usr/local/Cellar/cmake/3.3.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.3.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.3.2/share/cmake/Modules/FindOpenSSL.cmake:334 (find_package_handle_standard_args)
  src/CMakeLists.txt:62 (find_package)


thread '<main>' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', /Users/brendan/.multirust/toolchains/stable/cargo/registry/src/github.com-0a35038f75765ae4/cmake-0.1.11/src/lib.rs:401
```
I have openssl installed via brew:

```
λ brew info openssl
openssl: stable 1.0.2d (bottled)
OpenSSL SSL/TLS cryptography library
https://openssl.org/

This formula is keg-only.
Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries

/usr/local/Cellar/openssl/1.0.2d_1 (464 files, 18M)
  Poured from bottle
From: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/openssl.rb
==> Dependencies
Build: makedepend ✘
==> Options
--universal
	Build a universal binary
--without-check
	Skip build-time tests (not recommended)
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl/certs

and run
  /usr/local/opt/openssl/bin/c_rehash

This formula is keg-only, which means it was not symlinked into /usr/local.

Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries

Generally there are no consequences of this for you. If you build your
own software and it requires this formula, you'll need to add to your
build variables:

    LDFLAGS:  -L/usr/local/opt/openssl/lib
    CPPFLAGS: -I/usr/local/opt/openssl/include
```

Maybe there needs to be some validation in the build script to give a better error message for OS X users? This is an issue for installing things like `cargo-extras` easily, without doing much configuration.

--- a/libssh2-sys/build.rs
+++ b/libssh2-sys/build.rs
@@ -62,6 +62,11 @@ fn main() {
         }
     }
 
+    // Homebrew deprecated OpenSSL and deliberately hides it from cmake, requiring such opt-in
+    if target.contains("darwin") && Path::new("/usr/local/opt/openssl/include/openssl/ssl.h").exists() {
+        cfg.define("OPENSSL_ROOT_DIR", "/usr/local/opt/openssl/");
+    }
+
     let dst = cfg.define("BUILD_SHARED_LIBS", "OFF")
                  .define("ENABLE_ZLIB_COMPRESSION", "ON")
                  .define("CMAKE_INSTALL_LIBDIR", "lib")

